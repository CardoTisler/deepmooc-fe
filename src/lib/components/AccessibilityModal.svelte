<script lang="ts">
	import { Section } from '$lib/components';
	import theme from '$lib/shared/stores/theme';
	export let toggleMenu: () => void;

	const enableContrastMode = () => {
		theme.set('contrast');
	};
	const enableLightMode = () => {
		theme.set('light');
	};
	$: isContrastMode = $theme === 'contrast';
</script>

<Section class="w-full max-w-4xl p-4 {isContrastMode ? 'black-bg' : ''}">
	<div class="flex flex-col">
		<div class="flex justify-end">
			<!-- TODO: Replace with X icon -->
			<button class="p-2 focus:border-2 focus:border-blue-600" on:click={toggleMenu}>X</button>
		</div>
		<div>
			<h3 class="dark:text-yellow-300">Juurdepääsetavus</h3>
			<p class="dark:text-yellow-300">
				Veebileht on disainitud põhimõttega, et lehel olev informatsioon ning funktsionaalsused on
				ligipääsetavad võimalikult paljudele inimestele.
			</p>
			<br />
			<!-- <p>Loe rohkem <a href="#" class="underline">juurdepääsetavusest</a></p> -->
		</div>
		<div>
			<h3 class="dark:text-yellow-300">Teksti suurus</h3>
			<p class="dark:text-yellow-300">
				Enimkasutatavad veebibrauserid pakuvad võimalust suurendada või vähendada teksti suurust
				Ctrl nupu (Cmd nupp OS X operatsioonisüsteemi korral) all hoidmise ning + või - vajutamise
				kaudu. Teine võimalus on kasutada arvutihiirt: hoides all Ctrl nuppu ning liigutades
				rullikut samaaegselt. Sätete taastamiseks saab korraga alla vajutada Ctrl ja 0.
			</p>
			<br />
		</div>
		<div>
			<h3 class="dark:text-yellow-300">Kontrast</h3>
			<br />
			<div class="flex flex-row w-full xl:w-1/2 justify-between">
				<button
					class="text-xl border-2 border-primary-200 rounded-xl py-2 px-3 hover:bg-primary-200 dark:text-yellow-300 dark:border-yellow-300 dark:hover:bg-gray-900 focus:border-blue-600 focus:first"
					on:click={enableLightMode} autofocus>Tava sätted</button
				>
				<button
					class="text-xl focus:border-2 bg-gray-800 text-white rounded-xl py-2 px-3 focus:border-blue-600 hover:bg-gray-900 dark:text-yellow-300"
					on:click={enableContrastMode}
					>Must taust ja <span class="text-yellow-300">kollane tekst</span></button
				>
			</div>
		</div>
	</div>
</Section>

<style>
</style>
