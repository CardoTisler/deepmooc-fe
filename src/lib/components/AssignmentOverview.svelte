<script lang="ts">
	import Pill from '$lib/components/util/Pill.svelte';
	import type { AssignmentOverview } from '$lib/types';
	import RightArrow from './icons/RightArrow.svelte';
	export let data: AssignmentOverview;
	import theme from '$lib/shared/stores/theme';

	$: isContrastMode = $theme === 'contrast';
	const { assignmentId, topic, tags, title, deadlineTime, deadlineDate, completed, courseId } =
		data;
</script>

<!-- TODO: Need to accept and handle courseId for redirecting to courses/:courseId/assignments/:assignmentId -->
<div
	class="flex flex-col bg-gray-100 rounded-xl lg:h-36 p-2 border-gray-200 border-2 max-w-xl dark:bg-black dark:border-yellow-300"
>
	<div class="flex flex-col lg:flex-row justify-between">
		<div class=""><p>{topic}</p></div>
		<div class="flex flex-row justify-start lg:justify-end">
			{#each tags as tag}
				<div class="mx-1">
					<Pill text={tag} />
				</div>
			{/each}
		</div>
	</div>
	<div class="w-full py-2">
		<p class="text-xl font-bold">{title}</p>
	</div>
	<div class="flex flex-row w-full justify-between pt-3">
		<div class="flex flex-col">
			<p class="font-bold">{deadlineTime}</p>
			<p class="font-thin">{deadlineDate}</p>
		</div>
		<div class="flex items-center">
			{#if completed}
				<button
					class="border border-gray-700 text-gray-600 p-2 rounded-sm disabled dark:bg-gray-900"
					><span class="dark:text-yellow-300">Lahendatud</span></button
				>
			{:else}
				<a
					class="bg-primary-200 text-white hover:bg-primary-300 rounded-sm p-2 flex flex-row dark:bg-black dark:border-2 dark:border-yellow-300 dark:bg-gray-900 hover:dark:bg-black"
					href={`/courses/${courseId}/assignments/${assignmentId}`}
				>
					<span class="dark:text-yellow-300">Lahenda</span>
					<span class="pl-2">
						<RightArrow
							fill={`${isContrastMode ? 'yellow' : 'white'}`}
							stroke={`${isContrastMode ? 'yellow' : 'white'}`}
						/>
					</span>
				</a>
			{/if}
		</div>
	</div>
</div>

<style></style>
