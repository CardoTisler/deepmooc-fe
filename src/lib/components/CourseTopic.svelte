<script lang="ts">
	import { AssignmentOverview } from '.';
	import Section from './util/Section.svelte';
	let style = '';
	export { style as class };

	export let data: any;
</script>

<Section class={`flex flex-col p-8 max-w-2xl lg:max-w-5xl ${style}`}>
	<div>
		<p class="text-3xl text-semibold" aria-label="Teema pealkiri">{data.topicTitle}</p>
	</div>
	<div class="py-8" aria-label="Teema kirjeldus">
		<p>{data.topicDescription}</p>
	</div>
	<div class="flex flex-col" aria-label="Ülesanded">
		<p>ÜLESANDED</p>
		<div class="grid grid-cols-1 lg:grid-cols-2 gap-4">
			{#each data.assignments as assignment}
				<AssignmentOverview data={assignment} />
			{/each}
		</div>
	</div>
</Section>
