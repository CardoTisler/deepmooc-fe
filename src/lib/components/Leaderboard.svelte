<script lang="ts">
	import type { LeaderboardRow } from '$lib/types';
	export let rows: LeaderboardRow[];
	let style = '';
	export { style as class };
</script>

<div
	class="rounded-xl bg-gray-100 border-2 border-gray-200 h-36 flex flex-col justify-evenly container max-w-xl {style}"
>
	{#each rows as { index, name, score, bold }, i}
		<div class="grid grid-cols-3 gap-4 w-full">
			<div class="flex justify-start pl-4"><p class={bold ? 'font-bold' : ''}>{index}.</p></div>
			<div class="flex justify-center"><p class={bold ? 'font-bold' : ''}>{name}</p></div>
			<div class="flex justify-end pr-4"><p class={bold ? 'font-bold' : ''}>{score}</p></div>
		</div>
		{#if i == 0}
			<div class="grid grid-cols-3 gap-4 w-full">
				<div class="flex justify-start pl-4">
					<p>&#8942;</p>
				</div>
				<div class="flex justify-center">
					<p>&#8942;</p>
				</div>
				<div class="flex justify-end pr-4">
					<p>&#8942;</p>
				</div>
			</div>
		{/if}
	{/each}
</div>

<style>
	.container {
		min-width: 360px;
	}
</style>
