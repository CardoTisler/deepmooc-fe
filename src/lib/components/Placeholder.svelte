<script lang="ts"></script>

<div
	class="flex flex-col w-full border-dashed border-2 border-purple-400 text-center text-purple-400 align-center justify-center"
>
	<p class="block justify-center">Placeholder</p>
</div>

<style>
</style>
