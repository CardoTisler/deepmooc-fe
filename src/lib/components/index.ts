export { default as AssignmentOverview } from '$lib/components/AssignmentOverview.svelte';
export { default as CourseOverview } from '$lib/components/CourseOverview.svelte';
export { default as Leaderboard } from '$lib/components/Leaderboard.svelte';
export { default as Timeline } from "$lib/components/Timeline.svelte";
export { default as TimelineEvent } from "$lib/components/TimelineEvent.svelte";
