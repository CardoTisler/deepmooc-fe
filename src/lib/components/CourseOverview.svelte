<script lang="ts">
	import { AssignmentOverview, Leaderboard, Section } from '$lib/components';
	import type { CourseOverviewData } from '$lib/types';
	import RightArrow from './icons/RightArrow.svelte';
	import theme from '$lib/shared/stores/theme';

	$: isContrastMode = $theme === 'contrast';
	let style = '';
	export { style as class };

	export let data: CourseOverviewData;
	const { courseId, courseName, courseCode, leaderboardRows: rows, nextAssignment } = data;
</script>

<Section
	class={`p-8 grid grid-cols-1 gap-4 lg:gap-10 max-w-2xl lg:max-w-5xl lg:min-w-lg ${
		isContrastMode ? 'black-bg' : ''
	} ${style}`}
>
	<div class="flex flex-row justify-between" aria-label={`Kursuse ${courseName} ülevaade`}>
		<div>
			<h2 class="dark:text-yellow-300">{courseName}</h2>
			<p class="gray-spaced dark:text-yellow-300" aria-label={`Kursuse kood: ${courseCode}`}>{courseCode}</p>
		</div>
		<div class="ml-16 lg:ml-0 flex items-center">
			<a href={`/courses/${courseId}`} data-testid={`course-${courseId}`} aria-label={`Sisene kursusele ${courseName}`}
				><RightArrow
					fill={isContrastMode ? 'yellow' : 'black'}
					stroke={isContrastMode ? 'yellow' : 'black'}
				/></a
			>
		</div>
	</div>
	<div class="grid grid-cols-1 2xl:grid-cols-2 gap-8">
		<!-- TODO: What if there is no leaderboard? -->
		<div class="flex flex-col dark:text-yellow-300 dark:bg-black" aria-label="Edetabeli komponent">
			<p>EDETABEL</p>
			<Leaderboard {rows} />
		</div>
		<!-- TODO: What if there is no next assignment? -->
		<div class="flex flex-col dark:text-yellow-300" aria-label="Järgmise ülesande komponent">
			<p>JÄRGMINE ÜLESANNE</p>
			<AssignmentOverview data={nextAssignment} />
		</div>
	</div>
</Section>
