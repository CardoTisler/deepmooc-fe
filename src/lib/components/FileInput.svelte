<script lang="ts">
	import { Section } from '$lib/components';
	import theme from '$lib/shared/stores/theme';
	$: isContrastMode = $theme === 'contrast';
	let hasSelectedFile = false;
</script>

<div class="w-fit h-80 flex items-center justify-center dark:bg-black">
	<Section class="w-fit p-4 {isContrastMode ? 'black-bg' : ''}">
		<form action="?/assignment" method="POST" enctype="multipart/form-data">
			<label for="file-upload" class={isContrastMode ? 'label-bg-yellow' : 'label-bg-black'}
				>Upload file</label
			>
			<input
				id="file-upload"
				type="file"
				name="assignment"
				placeholder="assignment"
				class="input-file"
				on:change={() => {
					hasSelectedFile = true;
				}}
			/>
			{#if hasSelectedFile}
				<button
					type="submit"
					class="bg-primary-200 border hover:bg-primary-300 py-2 px-3 rounded-md dark:bg-black dark:text-yellow-300 dark:border-yellow-300 dark:hover:bg-gray-700"
				>
					<p>Lae fail Ã¼les</p>
				</button>
			{/if}
		</form>
	</Section>

	<!-- <Section class="w-1/2 p-4 h-fit">

    </Section> -->
</div>

<style>
	.input-file {
		width: 0.1px;
		height: 0.1px;
		opacity: 0;
		overflow: hidden;
		position: absolute;
		z-index: -1;
	}
	.label-bg-black {
		background: url('/export-button-svgrepo-com.svg');
		display: block;
		text-indent: -9999px;
		width: 100px;
		height: 82px;
		background-size: 100px 82px;
		cursor: pointer;
	}

	.label-bg-yellow {
		background: url('/export-button-svgrepo-com-yellow.svg');
		display: block;
		text-indent: -9999px;
		width: 100px;
		height: 82px;
		background-size: 100px 82px;
		cursor: pointer;
	}
	:global(.black-bg) {
		background: black;
	}
</style>
