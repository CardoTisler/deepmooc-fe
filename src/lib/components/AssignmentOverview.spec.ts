import { render } from '@testing-library/svelte';
import { AssignmentOverview } from '$lib/components';
import type { AssignmentOverview as IAssignmentOverview } from '$lib/types.ts';
import theme from '$lib/shared/stores/theme';

describe('AssignmentOverview.svelte', () => {
	it('should render component with assignment button enabled', () => {
		theme.set('light');
		const mockAssignment: IAssignmentOverview = {
			assignmentId: 1,
			courseId: 0,
			topic: 'TEST TOPIC',
			title: 'TEST TITLE',
			deadlineDate: '01.01.1970',
			deadlineTime: '00:00:00',
			tags: ['test tag'],
			completed: false
		};
		const { container } = render(AssignmentOverview, { data: mockAssignment });
		expect(container).toMatchSnapshot();
	});

	it('should render component with assignment button disabled', () => {
		theme.set('contrast');
		const mockAssignment: IAssignmentOverview = {
			assignmentId: 1,
			courseId: 0,
			topic: 'TEST TOPIC',
			title: 'TEST TITLE',
			deadlineDate: '01.01.1970',
			deadlineTime: '00:00:00',
			tags: ['test tag'],
			completed: true
		};
		const { container } = render(AssignmentOverview, { data: mockAssignment });
		expect(container).toMatchSnapshot();
	});
});
