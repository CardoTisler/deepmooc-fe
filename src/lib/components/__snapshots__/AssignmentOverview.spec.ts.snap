// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AssignmentOverview.svelte > should render component 1`] = `
<body>
  <div>
    <div
      class="flex flex-col bg-gray-100 rounded-xl lg:h-36 p-2 border-gray-200 border-2 max-w-xl dark:bg-black dark:border-yellow-300"
    >
      <div
        class="flex flex-col lg:flex-row justify-between"
      >
        <div
          class=""
        >
          <p>
            TEST TOPIC
          </p>
        </div>
         
        <div
          class="flex flex-row justify-start lg:justify-end"
        >
          <div
            class="mx-1"
          >
            <div
              class="rounded-full bg-gray-200 px-2 dark:bg-black "
            >
              <span
                class="text-gray-600 dark:text-yellow-300"
              >
                test tag
              </span>
            </div>
            <!--&lt;Pill&gt;-->
             
          </div>
        </div>
      </div>
       
      <div
        class="w-full py-2"
      >
        <p
          class="text-xl font-bold"
        >
          TEST TITLE
        </p>
      </div>
       
      <div
        class="flex flex-row w-full justify-between pt-3"
      >
        <div
          class="flex flex-col"
        >
          <p
            class="font-bold"
          >
            00:00:00
          </p>
           
          <p
            class="font-thin"
          >
            01.01.1970
          </p>
        </div>
         
        <div
          class="flex items-center"
        >
          <a
            class="bg-primary-200 text-white hover:bg-primary-300 rounded-sm p-2 flex flex-row dark:bg-black dark:border-2 dark:border-yellow-300 dark:bg-gray-900 hover:dark:bg-black"
            data-testid="assignment-1-solve-button"
            href="/courses/0/assignments/1"
          >
            <span
              class="dark:text-yellow-300"
            >
              Lahenda
            </span>
             
            <span
              class="pl-2"
            >
              <svg
                clip-rule="evenodd"
                fill="white"
                fill-rule="evenodd"
                height="24"
                stroke="white"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21.883 12l-7.527 6.235.644.765 9-7.521-9-7.479-.645.764 7.529 6.236h-21.884v1h21.883z"
                />
              </svg>
              <!--&lt;RightArrow&gt;-->
            </span>
          </a>
        </div>
      </div>
    </div>
    <!--&lt;AssignmentOverview&gt;-->
  </div>
</body>
`;
